### PR TITLE
azurerm_cosmosdb_cassandra_cluster - support new properties

### PR DESCRIPTION
**Pull request by neil-yechenwei**
This PR is to support new properties `authentication_method`, `client_certificate`, `external_gossip_certificate`, `external_seed_node`, `identity`, `repair_enabled` and `version`.

--- PASS: TestAccCassandraSequential/cluster/complete (873.82s)
--- PASS: TestAccCassandraSequential/cluster/update (1067.56s)
--- PASS: TestAccCassandraSequential/cluster/requiresImport (852.31s)
--- PASS: TestAccCassandraSequential/cluster/basic (830.39s)

### CODE REVIEW

File: website/docs/r/cosmosdb_cassandra_cluster.html.markdown [hunk 1]
```diff
@@ -77,11 +77,49 @@ The following arguments are supported:
...
+
+A `external_gossip_certificate` block supports the following:
+
+* `pem` - (Required) A list of TLS certificates that is used to authorize gossip from unmanaged Cassandra Data Center.
+
+---
+
+A `external_seed_node` block supports the following:
+
+* `ip_address` - (Required) A list of IP Addresses of the seed nodes in unmanaged the Cassandra Data Center which will be added to the seed node lists of all managed nodes.
```
Comment (katbyte) [inline website/docs/r/cosmosdb_cassandra_cluster.html.markdown:112]: would this make more sense as (also address should be addresses?)
```suggestion
A `external_seed_node_ip_addresses` - (Required) A list of IP Addresses of the seed nodes in unmanaged the Cassandra Data Center which will be added to the seed node lists of all managed nodes.
```

File: website/docs/r/cosmosdb_cassandra_cluster.html.markdown [hunk 2]
```diff
@@ -77,11 +77,49 @@ The following arguments are supported:
...
+
+A `client_certificate` block supports the following:
+
+* `pem` - (Required) A list of TLS certificates that is used to authorize client connecting to the Cassandra Cluster.
+
+---
+
+A `external_gossip_certificate` block supports the following:
+
+* `pem` - (Required) A list of TLS certificates that is used to authorize gossip from unmanaged Cassandra Data Center.
```
Comment (katbyte) [inline website/docs/r/cosmosdb_cassandra_cluster.html.markdown:106]: if its just pem then
```suggestion
A `client_certificate_pems` - (Required) A list of TLS certificates that is used to authorize client connecting to the Cassandra Cluster.

---

A `external_gossip_certificate_pems` - (Required) A list of TLS certificates that is used to authorize gossip from unmanaged Cassandra Data Center.
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @neil-yechenwei - comments left inline

File: website/docs/r/cosmosdb_cassandra_cluster.html.markdown (hunk 2 above)
Comment (neil-yechenwei) [reply website/docs/r/cosmosdb_cassandra_cluster.html.markdown:106]: If service team expands the sub properties for `external_seed_node`, seems currently it's not easily to support these new sub properties to TF. So suggest align with the property schema of service api. Does it make sense?

File: website/docs/r/cosmosdb_cassandra_cluster.html.markdown (hunk 1 above)
Comment (neil-yechenwei) [reply website/docs/r/cosmosdb_cassandra_cluster.html.markdown:112]: @katbyte , if service team expands the sub properties for `external_seed_node`, seems currently it's not easily to support these new sub properties to TF. So suggest align with the property schema of service api. Does it make sense?

Comment (katbyte) [reply website/docs/r/cosmosdb_cassandra_cluster.html.markdown:112]: do they have active plans to? it's easy enough to change it if we need to as there no bools here

File: website/docs/r/cosmosdb_cassandra_cluster.html.markdown (hunk 2 above)
Comment (katbyte) [reply website/docs/r/cosmosdb_cassandra_cluster.html.markdown:106]: do they have active plans to? it's easy enough to change it if we need to as there no bools here

File: website/docs/r/cosmosdb_cassandra_cluster.html.markdown (hunk 1 above)
Comment (neil-yechenwei) [reply website/docs/r/cosmosdb_cassandra_cluster.html.markdown:112]: I am just curious on why servic team treats it as block. I suppose they intend to expand sub properties in the future.

File: website/docs/r/cosmosdb_cassandra_cluster.html.markdown (hunk 2 above)
Comment (neil-yechenwei) [reply website/docs/r/cosmosdb_cassandra_cluster.html.markdown:106]: I am just curious on why servic team treats it as block. I suppose they intend to expand sub properties in the future.

**Comment by neil-yechenwei** [comment]
@katbyte , thanks for your comments. I've updated code. Please take another look. Thanks in advance.

File: website/docs/r/cosmosdb_cassandra_cluster.html.markdown (hunk 2 above)
Comment (neil-yechenwei) [reply website/docs/r/cosmosdb_cassandra_cluster.html.markdown:106]: Updated Code.

File: website/docs/r/cosmosdb_cassandra_cluster.html.markdown (hunk 1 above)
Comment (neil-yechenwei) [reply website/docs/r/cosmosdb_cassandra_cluster.html.markdown]: Updated Code.

**Comment by katbyte** [comment]
Lots of test failures
```
------- Stdout: -------
=== RUN   TestAccCassandraSequential
=== RUN   TestAccCassandraSequential/cluster
=== RUN   TestAccCassandraSequential/cluster/update
    testcase.go:117: Step 1/4 error: Error running apply: exit status 1
        
        Error: waiting on create for "Cassandra Cluster: (Name \"acctca-mi-cluster-220601014336782446\" / Resource Group \"acctest-ca-220601014336782446\")": Code="InternalServerError" Message="The resource failed to be created because of an internal configuration error: Conflict. Please file a support ticket.\r\nActivityId: c7938082-4962-45e2-bd33-c740f8d97fb1, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0"
        
          with azurerm_cosmosdb_cassandra_cluster.test,
          on terraform_plugin_test.tf line 37, in resource "azurerm_cosmosdb_cassandra_cluster" "test":
          37: resource "azurerm_cosmosdb_cassandra_cluster" "test" {
        
    testing_new.go:70: Error running post-test destroy, there may be dangling resources: exit status 1
        
        Error: deleting Subnet: (Name "acctsub-220601014336782446" / Virtual Network Name "acctvn-220601014336782446" / Resource Group "acctest-ca-220601014336782446"): network.SubnetsClient#Delete: Failure sending request: StatusCode=500 -- Original Error: Code="GatewayAuthenticationFailed" Message="Gateway authentication failed for 'Microsoft.Network'. Diagnostic information: timestamp '20220601T020647Z', tracking id '1dc5f9b6-41cd-45bb-a0a3-1227b19eb501', request correlation id 'fc6549e2-2be8-9183-769a-0aab9e829f17'."
        
=== RUN   TestAccCassandraSequential/cluster/requiresImport
=== RUN   TestAccCassandraSequential/cluster/basic
    testcase.go:117: Step 1/2 error: Error running apply: exit status 1
        
        Error: creating/updating Virtual Network: (Name "acctvn-220601022444882934" / Resource Group "acctest-ca-220601022444882934"): network.VirtualNetworksClient#CreateOrUpdate: Failure sending request: StatusCode=500 -- Original Error: Code="GatewayAuthenticationFailed" Message="Gateway authentication failed for 'Microsoft.Network'. Diagnostic information: timestamp '20220601T023254Z', tracking id '2c9443e0-5931-47a7-be87-4250a1ba8568', request correlation id 'fc6549e2-2be8-9183-769a-0aab9e829f17'."
        
          with azurerm_virtual_network.test,
          on terraform_plugin_test.tf line 12, in resource "azurerm_virtual_network" "test":
          12: resource "azurerm_virtual_network" "test" {
        
=== RUN   TestAccCassandraSequential/cluster/complete
    testcase.go:117: Step 1/2 error: Error running apply: exit status 1
        
```

**Comment by neil-yechenwei** [comment]
@katbyte , maybe it's just your environment issue. All TCs passed on my local and Teamcity.

The test result on my local:
--- PASS: TestAccCassandraSequential/cluster/requiresImport (912.15s)
--- PASS: TestAccCassandraSequential/cluster/basic (866.22s)
--- PASS: TestAccCassandraSequential/cluster/complete (872.29s)
--- PASS: TestAccCassandraSequential/cluster/update (1116.11s)

The test result on Teamcity:

**Review by katbyte — APPROVED** [review]
LGTM 🌵

**Pull request merged**